packages/alfa-rules/src/sia-r53/rule.ts

```diff
@@ -31,97 +32,84 @@ export default Rule.Atomic.of<Page, Element>({
         // * The target is in headings by construction of the applicability.
         // * The first element of heading is not a target due to the .skip(1)
         // * Therefore headings contain at least on element before the target.
-        const previous = headings.takeUntil(equals(target)).last().getUnsafe();
+        const previousHeading = headings
+          .takeUntil(equals(target))
+          .last()
+          .getUnsafe();
+
+        let previousLevel = -1;
+        let currentLevel = -1;
 
         return {
           1: expectation(
-            hasHeadingLevel(device, (currentLevel) =>
-              hasHeadingLevel(
-                device,
-                (previousLevel) => previousLevel >= currentLevel - 1,
-              )(previous),
+            hasHeadingLevel(
+              device,
+              tee(
+                (currentLevel) =>
+                  hasHeadingLevel(
+                    device,
+                    tee(
+                      (previousLevel) => previousLevel >= currentLevel - 1,
+                      (p) => {
+                        previousLevel = p;
+                      },
```

Comment:
**Nitpick:** I think we can have prettier inline that by using parenthesis instead of brackets 🤔